python/tk_multi_loader/delegate_publish_thumb.py

```diff
@@ -16,40 +17,176 @@
 shotgun_model = sgtk.platform.import_framework("tk-framework-shotgunutils", "shotgun_model")
 shotgun_view = sgtk.platform.import_framework("tk-framework-qtwidgets", "shotgun_view")
 
+from .ui.widget_publish_thumb import Ui_PublishThumbWidget
 
-class SgPublishDelegate(shotgun_view.WidgetDelegate):
+
+class PublishThumbWidget(QtGui.QWidget):
     """
-    Delegate which 'glues up' the ThumbWidget with a QT View.
+    Thumbnail style widget which contains an image and some 
+    text underneath. The widget scales gracefully. 
+    Used in the main loader view.
     """
+    
+    def __init__(self, parent):
+        """
+        Constructor
+        
+        :param parent: QT parent object
+        """
+        QtGui.QWidget.__init__(self, parent)
+
+        # make sure this widget isn't shown
+        self.setVisible(False)
+        
+        # set up the UI
+        self.ui = Ui_PublishThumbWidget() 
+        self.ui.setupUi(self)
+        
+        # set up an event filter to ensure that the thumbnails
+        # are scaled in a square fashion.
+        filter = ResizeEventFilter(self.ui.thumbnail)
```

Comment:
ResizeEventFilter is a really ambiguous name - maybe make it more specific or move it inside this class as a protected sub-class?